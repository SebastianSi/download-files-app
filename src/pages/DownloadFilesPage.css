.download-files-list {
  padding: initial;
  list-style-type: none;
}

.alert-box-file-text {
  font-size: large;
  color: rgba(255, 123, 0, 0.911);
}

ul.allowed .alert-box-file-text{
  color: rgb(5, 119, 5);
}