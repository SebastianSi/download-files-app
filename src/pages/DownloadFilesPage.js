import React, {useState} from 'react';

import {Modal} from '../common/components/Modal';
import { DownloadFilesTable } from '../containers/DownloadFilesTable';

import './DownloadFilesPage.css';

export const DownloadFilesPage = () => {
  const [showAlertBox, setShowAlertBox] = useState(false);
  const [alertBoxData, setAlertBoxData] = useState([]);
  
  const handleAlertBoxClose = () => {
    setShowAlertBox(false);
  };

  const handleDownloadFiles = (files) => {
    setShowAlertBox(true);
    setAlertBoxData(files);
  };

  const allowedFiles = alertBoxData?.filter(file => file.status?.props?.children?.includes('Available'));
  const unallowedFiles = alertBoxData?.filter(file => !file.status?.props?.children?.includes('Available'));

  const alertContent = (files => {
    return files.map((file, i) => 
      <li className='alert-box-file' key={i}>
        <strong className={`alert-box-file-text ${file.status}`}>{file.device} {file.path}</strong>
        {i!== files?.length -1  && <hr style={{width: '80%'}}/>}
      </li>
    )
  });

  return (
    <div >
      <DownloadFilesTable isBlurred={showAlertBox} onDownload={handleDownloadFiles}/>
      <Modal show={showAlertBox} onClose={handleAlertBoxClose}>
        <h2>Downloading</h2>
        <ul className='download-files-list allowed'>
          {alertContent(allowedFiles)}
        </ul>
        <br/>
        {
          Boolean(unallowedFiles?.length) &&
          <>
          <h4>Warning! The following files are not available for download:</h4>
          <ul className='download-files-list'>
            {alertContent(unallowedFiles)}
          </ul>
          <br/>
          </>
        }
      </Modal>
    </div>
  );
};