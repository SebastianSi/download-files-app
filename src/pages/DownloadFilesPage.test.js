import { render } from '@testing-library/react';

import { DownloadFilesPage } from './DownloadFilesPage';

describe('Download Files page', () => {
  it('matches the snapshot', () => {
    const { baseElement } = render(<DownloadFilesPage />);
    expect(baseElement).toMatchSnapshot();
  });
});
