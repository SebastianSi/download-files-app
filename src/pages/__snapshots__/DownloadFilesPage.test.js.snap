// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Download Files page matches the snapshot 1`] = `
<body>
  <div>
    <div>
      <div
        class="table-container "
      >
        <div
          class="table-header-actions"
        >
          <label>
            <input
              id="main-checkbox"
              type="checkbox"
            />
            None Selected
          </label>
          <div
            class="table-submit-component"
          >
            <svg
              fill="currentColor"
              height="1.2rem"
              role="img"
              title="Download selected files"
              viewBox="0 0 16 16"
              width="1.2rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M.5 9.9a.5.5 0 0 1 .5.5v2.5a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1v-2.5a.5.5 0 0 1 1 0v2.5a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2v-2.5a.5.5 0 0 1 .5-.5z"
              />
              <path
                d="M7.646 11.854a.5.5 0 0 0 .708 0l3-3a.5.5 0 0 0-.708-.708L8.5 10.293V1.5a.5.5 0 0 0-1 0v8.793L5.354 8.146a.5.5 0 1 0-.708.708l3 3z"
              />
            </svg>
            <span
              class="download-selected-text"
            >
              Download Selected
            </span>
          </div>
        </div>
        <table
          aria-label="Files Table"
        >
          <thead>
            <tr>
              <th
                scope="col"
              />
            </tr>
          </thead>
          <tbody />
          <tfoot />
        </table>
      </div>
    </div>
  </div>
</body>
`;
